api/db-cdr/generate-cdr/validate-prerequisites-exist.sh

```diff
@@ -14,17 +14,30 @@ CRITERIA_MENU="cb_criteria_menu.csv"
 PREP_CRITERIA="prep_criteria.csv"
 PREP_CRITERIA_ANCESTOR="prep_criteria_ancestor.csv"
 PREP_CLINICAL_TERMS="prep_clinical_terms_nc.csv"
-All_FILES=($CRITERIA_MENU $PREP_CRITERIA $PREP_CRITERIA_ANCESTOR $PREP_CLINICAL_TERMS)
+DATA_DICTIONARY_FILE="ds_data_dictionary.csv"
+All_FILES=($CRITERIA_MENU $PREP_CRITERIA $PREP_CRITERIA_ANCESTOR $PREP_CLINICAL_TERMS $DATA_DICTIONARY_FILE)
 INCOMPATIBLE_DATASETS=("R2019Q4R3" "R2019Q4R4")
 
+echo 'Check if all necessary file exists'
+for files in ${All_FILES[@]}; do
+  dataDictionaryExist=gs://$BUCKET/$BQ_DATASET/$CSV_HOME_DIR/$files
+  echo 'Check file' $files
+  gsutil -q stat $dataDictionaryExist
+  status=$?
+
+  if [[ $status == 0 ]]; then
+    echo 'File exists'
+  fi
+done
+
```

Comment:
This section of code isn't necessary. This code iterates All_FILES and since you added your csv files to this collection it will fail if it doesn't process it.

Reply:
Yes, however it was failing after processing the rest of the files. So in case we missed any one file, re-running would cause a manual cleanup. So i thought of adding another check at the start just so if anyone of the file is missing it will stop right there

Reply:
But were iterating the files twice.. doesn't make sense to do that

Reply:
Done